modules/promotion/src/Entity/Promotion.php

```diff
@@ -293,14 +310,26 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     $fields['end_date'] = BaseFieldDefinition::create('datetime')
       ->setLabel(t('End date'))
       ->setDescription(t('The date after which the promotion is invalid.'))
-      ->setRequired(TRUE)
+      ->setRequired(FALSE)
       ->setSetting('datetime_type', 'date')
       ->setDefaultValueCallback('Drupal\commerce_promotion\Entity\Promotion::getDefaultEndDate')
       ->setDisplayOptions('form', [
         'type' => 'datetime_default',
         'weight' => 4,
       ]);
 
+    $fields['has_end_date'] = BaseFieldDefinition::create('boolean')
```

Comment:
We don't need a field for this, we can infer it from whether the end_date field is empty.